firmware/boron-ins-fsm/src/consoleFunctions.cpp

```diff
@@ -221,56 +221,63 @@ int ins_threshold_set(String input){
-    EEPROM.get((ADDR_IM_DOORID+1),holder.byte2);  
-    EEPROM.get((ADDR_IM_DOORID+2),holder.byte3);  
+  // if echo, publish current door ID
+  if(*checkForEcho == 'e'){  
+    EEPROM.get(ADDR_IM_DOORID, doorIDHolder.byte3);  
+    EEPROM.get((ADDR_IM_DOORID + 1), doorIDHolder.byte2);  
+    EEPROM.get((ADDR_IM_DOORID + 2), doorIDHolder.byte1);  
 
-    char buffer[64];
     snprintf(buffer, sizeof(buffer), "{\"byte1\":\"%02X\", \"byte2\":\"%02X\", \"byte3\":\"%02X\"}", 
-            holder.byte1,holder.byte2,holder.byte3); 
-    Particle.publish("Current Door Sensor ID: ",buffer, PRIVATE);
+            doorIDHolder.byte3, doorIDHolder.byte2, doorIDHolder.byte1); 
+    Particle.publish("Current Door Sensor ID: ", buffer, PRIVATE);
+
+    // put door ID in buffer for return value
+    snprintf(buffer, sizeof(buffer), "%02X%02X%02X", doorIDHolder.byte3, doorIDHolder.byte2, doorIDHolder.byte1);
   } 
-  else //else not echo, so we have a new door ID to parse
+  else // else not echo, so we have a new door ID to parse
   {
-    //parse input string and update global door ID 
-    const char* byteholder1;
-    const char* byteholder2;
-    const char* byteholder3;
-    int split1 = command.indexOf(',');
-    byteholder1 = command.substring(0,split1).c_str();
-    globalDoorID.byte3 = (uint8_t)strtol(byteholder1,NULL,16);
-    int split2 = command.indexOf(',', split1+1);
-    byteholder2 = command.substring(split1+1,split2).c_str();
-    globalDoorID.byte2 = (uint8_t)strtol(byteholder2,NULL,16);
-    int split3 = command.indexOf(',', split2+1);
-    byteholder3 = command.substring(split2+1,split3).c_str();
-    globalDoorID.byte1 = (uint8_t)strtol(byteholder3,NULL,16);
+    // parse input string and update global door ID 
+    char* byteholder1;
+    char* byteholder2;
+    char* byteholder3;
+
+    int split1 = command.indexOf(',');                                // get index of first comma to delimit input
+    byteholder3 = strdup(command.substring(0, split1).c_str());       // get first byte of input and copy to holder variable
+    globalDoorID.byte3 = (uint8_t)strtol(byteholder3, NULL, 16);      // convert it to hex and set the third byte of the door ID
+
+    int split2 = command.indexOf(',', split1 + 1);
+    byteholder2 = strdup(command.substring(split1 + 1, split2).c_str());
+    globalDoorID.byte2 = (uint8_t)strtol(byteholder2, NULL, 16);
+
+    int split3 = command.indexOf(',', split2 + 1);
+    byteholder1 = strdup(command.substring(split2 + 1, split3).c_str());
+    globalDoorID.byte1 = (uint8_t)strtol(byteholder1, NULL, 16);
 
     //write new global door ID to flash
-    EEPROM.put(ADDR_IM_DOORID,globalDoorID.byte1);  
-    EEPROM.put((ADDR_IM_DOORID+1),globalDoorID.byte2);  
-    EEPROM.put((ADDR_IM_DOORID+2),globalDoorID.byte3);  
-  
-  } //end if-else
+    EEPROM.put(ADDR_IM_DOORID, globalDoorID.byte3);
+    EEPROM.put((ADDR_IM_DOORID + 1), globalDoorID.byte2);
+    EEPROM.put((ADDR_IM_DOORID + 2), globalDoorID.byte1);
```

Comment:
I'm concerned that this part is now backwards.

Given a door ID string "AB,CD,EF":
`split1`, `byteholder3`, and `globalDoorID.byte3` look like they will be `AB`.
`split2`, `byteholder2`, and `globalDoorID.byte2` look like they will be `CD`.
and
`split3`, `byteholder1`, and `globalDoorID.byte1` look like they will be `EF`.

which means I would have expected
ADDR_IM_DOORID = `EF` = `globalDoorID.byte1`
ADDR_IM_DOORID = `CD` = `globalDoorID.byte2`
ADDR_IM_DOORID = `AB` = `globalDoorID.byte3`

Am I reading it wrong? Have you tried this on a real Boron?